src/Controller/DownloadController.php

```diff
@@ -43,15 +43,15 @@ public function __construct(EntityHandler $entityHandler)
     /**
      * Produce json response for download dialog box.
      *
-     * @param string $id The id of the dataset to download.
+     * @param integer $id The id of the dataset to download.
      *
      * @Route("/download/{id}", name="pelagos_app_download_default")
      *
      * @return Response
      */
-    public function defaultAction(string $id)
+    public function defaultAction(int $id)
     {
-        $dataset = $this->entityHandler->get(Dataset::class, $id);
+        $dataset = $this->entityHandler->get(Dataset::class, (int) $id);
```

Comment:
Why cast to int, it should be an int, because of the type hint.

Reply:
It was debug code. Good catch